frontend/packages/operator-lifecycle-manager/src/components/modals/uninstall-operator-modal.tsx

```diff
@@ -136,35 +143,93 @@ export const UninstallOperatorModal: React.FC<UninstallOperatorModalProps> = ({
+    const uninstallOperator = () => {
+      const operatorUninstallPromises = [
+        k8sKill(SubscriptionModel, subscription, {}, deleteOptions),
+        ...(subscription?.status?.installedCSV
+          ? [
+              k8sKill(
+                ClusterServiceVersionModel,
+                {
+                  metadata: {
+                    name: subscription.status.installedCSV,
+                    namespace: subscription.metadata.namespace,
+                  },
                 },
-              },
-              {},
-              deleteOptions,
-            ),
-          ]
-        : []),
-      ...(removePlugins
-        ? [k8sPatch(ConsoleOperatorConfigModel, consoleOperatorConfig, [patch])]
-        : []),
-    ];
-    const operandDeletionPromises = deleteOperands
-      ? operands.map((operand: K8sResourceCommon) => {
-          const model = modelFor(referenceFor(operand));
-          return k8sKill(model, operand, {}, deleteOptions);
+                {},
+                deleteOptions,
+              ),
+            ]
+          : []),
+        ...(removePlugins
+          ? [k8sPatch(ConsoleOperatorConfigModel, consoleOperatorConfig, [patch])]
+          : []),
+      ];
+
+      handleOperatorUninstallPromise(Promise.all(operatorUninstallPromises))
+        .then(() => {
+          setOperatorUninstallFinished(true);
         })
-      : [];
-    if (operandDeletionPromises.length > 0) {
+        .catch(() => {
+          setOperatorUninstallFinished(true);
+        });
+    };
+
+    const setOperandErrorsTo = (
+      operators: K8sResourceCommon[],
+      errorMsg: string,
+    ): OperandError[] => {
+      return operators.reduce((acc: OperandError[], curr, i) => {
+        return acc.concat({
+          operand: operands[i],
+          errorMessage: errorMsg,
+        });
+      }, []);
+    };
+
+    const verifyDeletionOfOperands = () => {
+      const url = `${window.SERVER_FLAGS.basePath}api/list-operands?name=${subscriptionName}&namespace=${subscriptionNamespace}`;
+      let operandVerificationErrors: OperandError[] = [];
+      let intervalID: Timeout;
+
+      const finishVerification = () => {
+        clearInterval(intervalID);
+        setOperandsDeleteInProgress(false);
+        setOperandsDeleteFinished(true);
+        if (operandVerificationErrors.length === 0) {
+          uninstallOperator();
+        } else {
+          // show operand deletion verification errors on results page
+          setOperandDeletionErrors(operandVerificationErrors);
+          setOperatorUninstallFinished(true);
+        }
+      };
+
+      intervalID = setInterval(() => {
+        coFetchJSON(url)
+          .then((curOperands) => {
+            setNumberOperandsRemaining(curOperands.items.length);
+            if (curOperands.items.length === 0) {
+              setTimeout(() => finishVerification(), 500); // allow '0 Operands remaining' to display for half a second
```

Comment:
Wondering if the 500milsec. is enough, since it will just cause the '0 Operands remaining' to flicker in the modal.
Watching at the recording I though I've missed something important since the msg was there displayed for such a short period of time.
I would either put it to 2 seconds or wait for users interaction by hitting the "OK" [button](https://github.com/openshift/console/blob/master/frontend/packages/operator-lifecycle-manager/src/components/modals/uninstall-operator-modal.tsx#L290)

Reply:
Ok, I felt the time was enough for user to see the progress bar goto 100% (all blue bar), I'll increase it.

Reply:
updated to a second.  video is a little old, it transitions from the progres bar, once at 100%, changes to screen which says "Uninstalling Operator...", then if no errors, closes modal automatically.  If operands are deleted quickly and no errors uninstalling the operator modal closes rather quickly.